src/test/java/org/broadinstitute/hellbender/tools/spark/sv/SVKmerShortUnitTest.java

```diff
@@ -28,40 +30,152 @@ public void testDefaultConstructionWithTooSmallK() {
 
     @DataProvider(name = "sequenceStrings")
     public Object[][] getSequenceStrings() {
-        return new Object[][] {
+        return new Object[][]{
                 {"ACGTACGTACGT"}, {"ACGTACGTACGTC"}, {"ACGTACGTACGTCC"}, {"ACGTACGTACGTCCC"}, {"ACGTACGTACGTCCCC"}
         };
     }
 
     @Test(dataProvider = "sequenceStrings")
-    public void testConstructionAndToString( final String str ) {
-        Assert.assertEquals(str, SVKmerizer.toKmer(str,new SVKmerShort(str.length())).toString(str.length()));
-        Assert.assertEquals(str, SVKmerizer.toKmer(str.getBytes(),new SVKmerShort(str.length())).toString(str.length()));
+    public void testConstructionAndToString(final String str) {
+        Assert.assertEquals(str, SVKmerizer.toKmer(str, new SVKmerShort(str.length())).toString(str.length()));
+        Assert.assertEquals(str, SVKmerizer.toKmer(str.getBytes(), new SVKmerShort(str.length())).toString(str.length()));
     }
 
     @Test(dataProvider = "sequenceStrings")
-    public void testSuccessor( final String str ) {
+    public void testSuccessor(final String str) {
         final int K = str.length();
         final StringBuilder sb = new StringBuilder(str);
         sb.delete(0, 1);
         sb.append('A');
-        final SVKmer kkk = SVKmerizer.toKmer(str,new SVKmerShort(str.length()));
+        final SVKmer kkk = SVKmerizer.toKmer(str, new SVKmerShort(str.length()));
         Assert.assertEquals(sb.toString(), kkk.successor(SVKmerLong.Base.A, K).toString(K));
-        sb.setCharAt(K-1, 'C');
+        sb.setCharAt(K - 1, 'C');
         Assert.assertEquals(sb.toString(), kkk.successor(SVKmerLong.Base.C, K).toString(K));
-        sb.setCharAt(K-1, 'G');
+        sb.setCharAt(K - 1, 'G');
         Assert.assertEquals(sb.toString(), kkk.successor(SVKmerLong.Base.G, K).toString(K));
-        sb.setCharAt(K-1, 'T');
+        sb.setCharAt(K - 1, 'T');
         Assert.assertEquals(sb.toString(), kkk.successor(SVKmerLong.Base.T, K).toString(K));
     }
 
+    @Test
+    public void testMaskedKmers() {
+
+        final int K = 31;
+        final RandomDNA randDNA = new RandomDNA(379483L);
+        final int seqLen = 1000;
+        final byte[] baseBytes = randDNA.nextBases(seqLen);
+        final String bases = new String(baseBytes);
+        final Random rand = new Random(2738493L);
+        for (int i = 0; i < seqLen - K; i++) {
+
+            //Random k-mer
+            String substr = bases.substring(i, i + K);
+
+            //For generating the correct answer
+            final StringBuilder sb = new StringBuilder(substr);
+
+            //Generate random indices to mask, an even number of them so that the k-mer will still be odd
```

Comment:
I think there's going to be a problem with canonicalization if you happen to choose the middle base as one of the ones to be masked.